src/components/MultipleSelectionDialog/MultipleSelectionDialog.tsx

```diff
@@ -5,95 +5,47 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import {
-    Button,
-    Checkbox,
-    Dialog,
-    DialogActions,
-    DialogContent,
-    DialogTitle,
-    FormControlLabel,
-    Grid,
-    List,
-} from '@mui/material';
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from '@mui/material';
 import { FormattedMessage } from 'react-intl';
 import { useState } from 'react';
+import { CheckboxList } from '../CheckBoxList/check-box-list';
 
-export interface MultipleSelectionDialogProps {
-    options: string[];
-    selectedOptions: string[];
+export interface MultipleSelectionDialogProps<T> {
+    options: T[];
+    selectedOptions: T[];
     open: boolean;
-    getOptionLabel: (option: string) => string;
+    getOptionLabel: (option: T) => string;
+    getOptionId: (option: T) => string;
     handleClose: () => void;
-    handleValidate: (ids: string[]) => void;
+    handleValidate: (options: T[]) => void;
     titleId: string;
```

Comment:
make MultipleSelectionDialogProps extend CheckboxListProps<T>, this way, all props from CheckboxList can be used in MultipleSelectionDialog